packages/sdk/src/browser-wallets/xverse/index.ts

```diff
@@ -40,10 +118,72 @@ async function getAddresses(
  * @returns An object containing `base64` and `hex` if the transaction is not extracted, or `hex` if the transaction is extracted.
  */
 async function signPsbt(
-  _psbt: Psbt,
-  _options: BrowserWalletSignPSBTOptions = {},
+  psbt: Psbt,
+  {
+    finalize = true,
+    extractTx = true,
+    network,
+    inputsToSign,
+  }: XverseSignPSBTOptions = { network: "mainnet", inputsToSign: [] },
 ): Promise<BrowserWalletSignResponse> {
-  throw new OrditSDKError("Method not implemented");
+  if (!isInstalled()) {
+    throw new BrowserWalletNotInstalledError("Xverse not installed");
+  }
+  if (!psbt || !network || !inputsToSign.length) {
+    throw new OrditSDKError("Invalid options provided");
+  }
+
+  let hex: string;
+  let base64: string | null = null;
+
+  const handleOnFinish = (response: SignTransactionResponse) => {
+    const { psbtBase64 } = response;
+    if (!psbtBase64) {
+      throw new BrowserWalletUserCancelledError("Request canceled by user.");
+    }
+
+    const signedPsbt = Psbt.fromBase64(psbtBase64);
+    if (finalize) {
+      if (!inputsToSign.length) {
+        signedPsbt.finalizeAllInputs();
+      } else {
+        inputsToSign.forEach((input) => {
+          input.signingIndexes.forEach((index) => {
+            signedPsbt.finalizeInput(index);
+          });
+        });
+      }
+    }
+
+    hex = extractTx
+      ? signedPsbt.extractTransaction().toHex()
+      : signedPsbt.toHex();
+    base64 = !extractTx ? signedPsbt.toBase64() : null;
+  };
+
+  const handleOnCancel = () => {
+    throw new BrowserWalletUserCancelledError(
+      `Failed to sign transaction using xVerse`,
+    );
+  };
+
+  const xverseOptions: SignTransactionOptions = {
+    payload: {
+      network: {
+        type: fromBrowserWalletNetworkToBitcoinNetworkType(network),
+      },
+      message: "Sign transaction",
```

Comment:
same here, use technically is signing `psbt`.

Reply:
Changed!